java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/cdk/TearDownMavenSystemTest.java

```diff
@@ -111,6 +129,9 @@ public static void tearDown(
                     List.of(buildSystemTestECRRepoName(shortId)));
         }
         LOGGER.info("Tear down finished, took {}", LoggedDuration.withFullOutput(startTime, Instant.now()));
+        if (!tearDownFailures.isEmpty()) {
+            throw new TearDownInstancesException(tearDownFailures);
+        }
```

Comment:
Is there value in gathering all the exceptions like this? It's a little difficult to understand the implications of it. If there's a failure that should fail the whole script, maybe we should just stop everything straight away and throw the exception? In that case we'll still have proper logging, and a non-zero exit code.